packages/xo-server/src/api/xosan.js

```diff
@@ -1138,9 +1139,14 @@ async function _prepareGlusterVm (
   await xapi.startVm(newVM)
   log.debug(`waiting for boot of ${ip}`)
   // wait until we find the assigned IP in the networks, we are just checking the boot is complete
-  const vmIsUp = vm =>
-    Boolean(vm.$guest_metrics && includes(vm.$guest_metrics.networks, ip))
-  const vm = await xapi._waitObjectState(newVM.$id, vmIsUp)
+  // fix #3688
+  const vm = await xapi._waitObjectState(
+    newVM.$id,
+    _ => _.guest_metrics !== NULL_REF
+  )
+  await xapi._waitObjectState(newVM.guest_metrics, _ =>
```

Comment:
AFAIU, `newVM` is an XO object therefore it has no `guest_metrics` property, you want to use `vm.guest_metrics` instead.